### ISSUE TITLE
Configuring TRD w/ Docker (instructions unclear)

### ISSUE THREAD

**Issue by mathee06** [tezos-reward-distributor-organization/tezos-reward-distributor#416]
**Describe the bug**
Unable to launch the TRD configuration wizard under the Tezos Docker environment. Tezos-Signer is running as a docker container and is able to send transactions out from the node.

**To Reproduce**
We have tried the following variations of executing the configure.py script, but have been unsuccessful in the Docker environment.
- python3 src/configure.py
- python3 src/configure.py -d
- python3 src/configure.py -d -E ../../mainnet.sh
- python3 src/configure.py -d -E http://0.0.0.0:9732
- python3 src/configure.py -d -E http://127.0.0.1:9732

For these commands, we have been receiving some variation of the following error:
```
2021-04-14 14:06:17,116 - MainThread - INFO - Arguments Configuration = {
 "reward_data_provider": "prpc",
 "network": "MAINNET",
 "reports_base": "~/pymnt/reports",
 "config_dir": "~/pymnt/cfg",
 "node_endpoint": "http://127.0.0.1:8732",
 "signer_endpoint": "http://127.0.0.1:6732",
 "docker": true,
 "verbose": "on",
 "dry_run": false,
 "api_base_url": null,
 "log_file": "logs/app.log"
}
2021-04-14 14:06:17,120 - MainThread - DEBUG - Error, request ->http://127.0.0.1:8732/chains/main/blocks/head/context/constants<-, params ->None<-,
---
Error, exception ->HTTPConnectionPool(host='127.0.0.1', port=8732): Max retries exceeded with url: /chains/main/blocks/head/context/constants (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff2901efc40>: Failed to establish a new connection: [Errno 111] Connection refused'))<-
2021-04-14 14:06:17,121 - MainThread - DEBUG - Error, request ->http://127.0.0.1:8732/chains/main/blocks/head/context/constants<-, params ->None<-,
---
Error, exception ->HTTPConnectionPool(host='127.0.0.1', port=8732): Max retries exceeded with url: /chains/main/blocks/head/context/constants (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff29023e6a0>: Failed to establish a new connection: [Errno 111] Connection refused'))<-
2021-04-14 14:06:17,122 - MainThread - DEBUG - Failed to get network configuration constants from a local node (http://127.0.0.1:8732).
2021-04-14 14:06:17,779 - MainThread - DEBUG - Network configuration constants successfully loaded from a public node (https://mainnet-tezos.giganode.io).
2021-04-14 14:06:17,779 - MainThread - DEBUG - Network config {'NB_FREEZE_CYCLE': 5, 'BLOCK_TIME_IN_SEC': 60, 'BLOCKS_PER_CYCLE': 4096, 'BLOCKS_PER_ROLL_SNAPSHOT': 256, 'BLOCK_REWARD': 40000000, 'ENDORSEMENT_REWARD': 1250000, 'NAME': 'MAINNET'}
2021-04-14 14:06:40,289 - MainThread - DEBUG - RpcBlockApiImpl - node_url http://127.0.0.1:8732
2021-04-14 14:06:41,524 - MainThread - DEBUG - Error, request ->http://127.0.0.1:6732/keys/tz1PeyZkxMBgbVi<-, params ->None<-,
---
Error, exception ->HTTPConnectionPool(host='127.0.0.1', port=6732): Max retries exceeded with url: /keys/tz1PeyZkxMBgbVi (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff29024b2e0>: Failed to establish a new connection: [Errno 111] Connection refused'))<-
2021-04-14 14:06:41,525 - MainThread - DEBUG - Error, request ->http://127.0.0.1:6732/keys/tz1PeyZkxMBgbVi<-, params ->None<-,
---
Error, exception ->HTTPConnectionPool(host='127.0.0.1', port=6732): Max retries exceeded with url: /keys/tz1PeyZkxMBgbVi (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7ff2901e0f10>: Failed to establish a new connection: [Errno 111] Connection refused'))<-
```

**Screenshots**
```
t3zos@tezos-node:~/tezos-docker$ docker ps
CONTAINER ID   IMAGE                  COMMAND                  CREATED       STATUS       PORTS                              NAMES
fbc3e14b5e4d   tezos-mainnet-signer   "/usr/local/bin/entr…"   3 hours ago   Up 3 hours                                      focused_dijkstra
_<... other containers>_
8ef5e7e9e7a8   tezos/tezos:v8.2       "/usr/local/bin/entr…"   4 weeks ago   Up 4 days    8732/tcp, 0.0.0.0:9732->9732/tcp   mainnet_node_1
```

**Environment**
 - Docker running on Ubuntu OS
 - mainnet.sh file is found two folders up from the tezos-reward-distributor repo (../../mainnet.sh)

**Comment by mathee06** [comment]
Hi @amzid - just following up. I am just looking to run the TRD configure.py wizard while using the Docker build of Tezos. The instructions are unclear how this should be done, I can submit a PR after the fact to correct it if needed. Please let me know.

**Comment by utdrmac** [comment]
Hi @mathee06 We don't have a docker image for TRD. How are you trying to run TRD this way? You can try building a Dockerfile yourself using one of the official python3 images and then cloning TRD into that maybe?

I do see in your docker ps output that your signer container does not have any ports mapped. Your signer should be running in http mode and the primary port should be forwarded into it. Also, I recommend to make a new network and launch all tezos related containers within that network for easy DNS resolution.

**Comment by mathee06** [comment]
Hi @utdrmac appreciate the help. 

I am using the docker version of Tezos - so I'm looking to have the regular TRD (installed on host) interact with the docker containers of Tezos.

Is this possible yet? _The instructions imply so at the moment_ - but that is whats unclear since the documentation stops midway (as I realized too late 😅).

Loading TRD in a Python container is smart and something I had not thought of - but I would have hoped there was a way to get TRD to recognize/communicate with the Tezos docker containers (out of the box?)

Thank you,

Mathee

**Comment by mathee06** [comment]
Also RE: _I do see in your docker ps output that your signer container does not have any ports mapped. Your signer should be running in http mode and the primary port should be forwarded into it. Also, I recommend to make a new network and launch all tezos related containers within that network for easy DNS resolution._

You are correct - a new network would be ideal however I was following instructions from the TRD documentation here: https://tezos-reward-distributor-organization.github.io/tezos-reward-distributor/tezossignerdocker.html

**Comment by utdrmac** [comment]
@mathee06 TRD only needs communication with tezos-signer and some source of reward data (node, tzstats, tzkt). So if you have tezos-signer running in a container, simply port forward whatever port you have signer running and point TRD to that cointainer's IP.

Both the `-d` and that docker documentation are old/out-of-date/not-maintained. TRD used to require direct access to the tezos-client program, but we have since removed that and now use tezos-signer over http.

**Comment by jdsika** [comment]
solved by #442
